core/src/apps/bitcoin/sign_tx/helpers.py

```diff
@@ -291,6 +334,8 @@ def sanitize_tx_input(txi: TxInput, coin: CoinInfo) -> TxInput:
             raise wire.DataError("Segwit not enabled on this coin.")
     if txi.commitment_data and not txi.ownership_proof:
         raise wire.DataError("commitment_data field provided but not expected.")
+    if txi.orig_hash and txi.orig_index is None:
+        raise wire.DataError("Missing orig_index field.")
```

Comment:
After #1297 is merged, I can see two ways to integrate the experimental features setting into this PR:
1. Check `wire.experimental_enable` here when `txi.orig_hash` is set. The same applies to `sanitize_tx_output()` below.
2. Extend `google.protobuf.FieldOptions` with an `experimental` flag, add `FLAG_EXPERIMENTAL` in `protobuf.py` and handle the check in `load_message()`. It seems like we would need to pass `wire.experimental_enable` as a parameter to `load_message()`.

Any preference @matejcik?

Reply:
I tried to implement Solution 2 (e69c74fc) and it seems to be working well. The only problem is that it doesn't support repeated experimental fields, but we don't need those currently.

Reply:
It also doesn't support required experimental fields, or experimental fields with default values. This is the limitation of re-using the "default" descriptor value for flags; if we wanted to do this properly, we would have to change that.

OTOH maybe that's not necessary?
It doesn't make sense to have a required experimental field, because the host is definitely not guaranteed to send it.
I'm not entirely sure about experimental field with a default value; but our current semantic for experimental fields is "the host sends the value in order to enable experimental behavior". So having it replaced by a default is same as if the host explicitly enabled the behavior; which makes little sense.

At least the protobuf generator should fail in case one of the unsupported combinations arises.

Reply:
Done https://github.com/trezor/trezor-firmware/commit/c76358ea6a644fa233e141b917d2e08971a7e2aa

Reply:
any reason to keep the commit outside this PR?

Reply:
Actually I think not. I just wanted to experiment outside of the branch. Cherry-picked: https://github.com/trezor/trezor-firmware/pull/1292/commits/8c39417593768375fc9af95361606a4137de6365 https://github.com/trezor/trezor-firmware/pull/1292/commits/d44a7453f67499810adb70ff7b49bcb8cb98a53b